src/html5_parser/tokenizer/mod.rs

```diff
@@ -2271,16 +2272,22 @@ impl<'a> Tokenizer<'a> {
     }
 
     // Sets the given name into the current token
-    fn set_name_in_current_token(&mut self, new_name: String) {
-        match &mut self.current_token.as_mut().unwrap() {
+    fn set_name_in_current_token(&mut self, new_name: String) -> Result<()> {
+        match &mut self.current_token.as_mut().expect("current token") {
             Token::StartTagToken { name, .. } => {
                 *name = new_name;
             }
             Token::EndTagToken { name, .. } => {
                 *name = new_name;
             }
-            _ => panic!("trying to set the name of a non start/end tag token"),
+            _ => {
+                return Err(Error::Parsing(
+                    "trying to set the name of a non start/end tag token".into(),
+                ))
```

Comment:
This is just an example of the kind of error handling we can introduce as an alternative to panicking.